server/src/server.rs

```diff
@@ -620,54 +823,81 @@ impl<HttpMiddleware, RpcMiddleware> Builder<HttpMiddleware, RpcMiddleware> {
 			listener,
 			cfg: self.settings,
 			rpc_middleware: self.rpc_middleware,
-			id_provider: self.id_provider,
 			http_middleware: self.http_middleware,
 		})
 	}
 }
 
 /// Data required by the server to handle requests.
 #[derive(Debug, Clone)]
-pub(crate) struct ServiceData {
+pub struct ServiceData {
```

Comment:
Is this still exposed publically or can the `pub`s now be removed?

Offhand it looks like it's only configured internally again now but I might be missing something!

Reply:
nice catch, it should be fixed now